overlord/snapshotstate/snapshotstate_test.go

```diff
@@ -1757,3 +1767,128 @@ func (snapshotSuite) TestEstimateSnapshotSizeWithUsers(c *check.C) {
 	c.Assert(err, check.IsNil)
 	c.Check(gotUsers, check.DeepEquals, []string{"user1", "user2"})
 }
+
+func (snapshotSuite) TestExportSnapshotConflictsWithForget(c *check.C) {
+	st := state.New(nil)
+	st.Lock()
+	defer st.Unlock()
+
+	chg := st.NewChange("forget-snapshot-change", "...")
+	tsk := st.NewTask("forget-snapshot", "...")
+	tsk.SetStatus(state.DoingStatus)
+	tsk.Set("snapshot-setup", map[string]int{"set-id": 42})
+	chg.AddTask(tsk)
+
+	_, err := snapshotstate.Export(context.TODO(), st, 42)
+	c.Assert(err, check.NotNil)
+	c.Assert(err.Error(), check.Equals, `cannot operate on snapshot set #42 while change "1" is in progress`)
+}
+
+func (snapshotSuite) TestImportSnapshotDuplicatedNoConflict(c *check.C) {
+	buf := &bytes.Buffer{}
+	var importCalls int
+	restore := snapshotstate.MockBackendImport(func(ctx context.Context, id uint64, r io.Reader, flags *backend.ImportFlags) ([]string, error) {
+		importCalls++
+		c.Check(id, check.Equals, uint64(1))
+		// FIXME: DuplicatedSnapshotImportError should include snap names
```

Comment:
There is a bug there in the existing code, I'll try to address it separately because the fix might be slightly annoying (and would grow this PR).

Reply:
This is addressed in #9851 , I'll update this PR when 9851 lands.